store/store_test.go

```diff
@@ -1707,6 +1925,52 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesAuth(c
 	c.Check(refreshDischargeEndpointHit, Equals, true)
 }
 
+func (t *remoteRepoTestSuite) TestUbuntuStoreRepositoryAssertionRefreshesDeviceAuth(c *C) {
```

Comment:
It is just to be sure the different methods hitting the store server are refresh-ready. I doubted about refactoring and adding tests for the doRequest method instead, testing the support against it only. Maybe for a future PR?